File: src/main/java/com/leijendary/spring/boot/template/core/config/PropertiesConfiguration.java
package com.leijendary.spring.boot.template.core.config;

import com.leijendary.spring.boot.template.core.config.properties.AuthProperties;
import com.leijendary.spring.boot.template.core.config.properties.InfoProperties;

import org.springframework.boot.context.properties.EnableConfigurationProperties;
import org.springframework.context.annotation.Configuration;

@Configuration
@EnableConfigurationProperties({ AuthProperties.class, InfoProperties.class })
public class PropertiesConfiguration {
}

File: src/main/kotlin/com/leijendary/spring/boot/template/core/config/PropertiesConfiguration.kt
package com.leijendary.spring.boot.template.core.config

import com.leijendary.spring.boot.template.core.config.properties.*
import org.springframework.boot.context.properties.EnableConfigurationProperties
import org.springframework.context.annotation.Configuration

@Configuration
@EnableConfigurationProperties(
    AuthProperties::class,
    DataSourcePrimaryProperties::class,
    DataSourceReadonlyProperties::class,
    InfoProperties::class,
    NumberProperties::class,
    RetryProperties::class
)
class PropertiesConfiguration 